### DISCUSSION TITLE
Using MEGALINTER_FILES_TO_LINT

### DISCUSSION THREAD

**Discussion by donk-msft** [oxsecurity/megalinter#2924] [Q&A] [answered]
I'm running Mega Linter is a separate repo and clone the remote repo to check during runtime. Currently, the validation works fine, but Megalinter processes all files in the repo, not just the ones that are changed.

I can't use VALIDATE_ALL_CODEBASE because I'm pulling the code from another repo. An alternative approach seems to be the option MEGALINTER_FILES_TO_LINT. I pass a string of files to lint and I should be good. For example, I tried to pass 'scripts/Update-Variables.ps1' for MEGALINTER_FILES_TO_LINT.

However,  when Megalinter starts linting, it looks for the files I passed in the /tmp/lint folder and cannot find them.  Below is a snippet from the log that shows Megalinter is adding /tmp/lint to the search path: 
```
MegaLinter now collects the files to analyse (expand for details)
[File listing] Input file /tmp/lint/scripts/Update-Variables.ps1 not found
```
How can I make sure Megalinter looks for the files in /CheckMe?

My code is:
```
# Run MegaLinter
- script: |
    docker run -v $(System.DefaultWorkingDirectory):/tmp/lint \
      --env-file <(env | grep -e SYSTEM_ -e BUILD_ -e TF_ -e AGENT_) \
      -e SYSTEM_ACCESSTOKEN=$(System.AccessToken) \
      -e GIT_AUTHORIZATION_BEARER=$(System.AccessToken) \
      -e MEGALINTER_FILES_TO_LINT='$(ChangedFiles)' \
      oxsecurity/megalinter:v7
  displayName: Run MegaLinter
```

The folder structure on my agent looks like this:

├── CheckMe
│   ├── README.md
│   ├── scripts
│   │   └── **Update-Variables.ps1**
│   └── src
├── pipelines
│   └── megaLinter.yml
└── src
    └── pssa
        └── Measure-Case.psm1

**Comment by Kurt-von-Laven** [comment, marked as answer]
I haven't tried using MegaLinter that way, but my first instinct would be to set [`DEFAULT_WORKSPACE: /tmp/lint/CheckMe`](https://megalinter.io/latest/config-variables/).

**Comment by donk-msft** [comment]
Thank Kurt, that did the trick. I was looking at it the wrong way and tried /CheckMe. 

Using /tmp/lint/Checkme works, but the downside is that all config files and custom rule sets in the root of the repo cannot be found anymore and need to be copied to the the root of the CheckMe folder.

Alternative is to add Checkme/ to every file to be checked instead.

**Comment by nvuillam** [comment]
@GitHubUser1339 can you try with beta version ?
I recently fixed a bug that has not been released yet, it could do the trick :)

-> https://github.com/oxsecurity/megalinter/issues/2744

**Reply by Kurt-von-Laven** [reply]
My understanding is that the expected behavior on beta is that `DEFAULT_WORKSPACE: CheckMe` works and `DEFAULT_WORKSPACE: /tmp/lint/CheckMe` doesn't?

**Comment by donk-msft** [comment]
I'm currently using the /tmp/lint as default workspace (e.g. I don't specify) and I add the 'CheckMe/' prefix to the changed files path I'd like to check. This setup works for me.
